idaes/models_extra/gas_solid_contactors/unit_models/fixed_bed_1D.py

```diff
@@ -0,0 +1,2067 @@
+                gas_phase.reaction_package.rate_reaction_idx,
+                doc="Gas side rate reaction extent",
+            )
+            def gas_phase_config_rxn_ext(b, t, x, r):
+                return b.gas_phase.rate_reaction_extent[t, x, r] == (
+                    b.gas_phase.reactions[t, x].reaction_rate[r]
+                    * b.gas_phase.area[t, x]
+                )
+
+        # Build hetereogeneous reaction constraints
+        if solid_phase.reaction_package is not None:
+            # Gas side heterogeneous rate reaction generation
+            @self.Constraint(
+                self.flowsheet().time,
+                self.length_domain,
+                gas_phase.property_package.phase_list,
+                gas_phase.property_package.component_list,
+                doc="Gas side heterogeneous" "rate reaction generation",
+            )
+            def gas_comp_hetero_rxn(b, t, x, p, j):
+                return b.gas_phase.mass_transfer_term[t, x, p, j] == (
+                    sum(
+                        b.solid_reactions[t, x].rate_reaction_stoichiometry[r, p, j]
+                        * b.solid_reactions[t, x].reaction_rate[r]
+                        for r in (solid_phase.reaction_package.rate_reaction_idx)
+                    )
+                    * b.solid_phase_area[t, x]
+                )
+
+        # ---------------------------------------------------------------------
+        # Solid phase component balance
+
+        # material holdup constraint
+        @self.Constraint(
+            self.flowsheet().config.time,
+            self.length_domain,
+            solid_phase.property_package.component_list,
+            doc="Solid phase material holdup constraints",
+        )
+        def solid_material_holdup_calculation(b, t, x, j):
+            return b.solid_material_holdup[t, x, j] == (
+                b.solid_phase_area[t, x]
+                * b.solid_properties[t, x].get_material_density_terms("Sol", j)
+            )
+
+        # Add component balances
+        @self.Constraint(
+            self.flowsheet().time,
+            self.length_domain,
+            solid_phase.property_package.component_list,
+            doc="Material balances",
+        )
+        def solid_material_balances(b, t, x, j):
+            if solid_phase.reaction_package is not None:
+                return b.solid_material_accumulation[t, x, j] == (
+                    b.solid_phase_area[t, x]
+                    * b.solid_properties[t, x]._params.mw_comp[j]
+                    * sum(
+                        b.solid_reactions[t, x].reaction_rate[r]
+                        * solid_phase.reaction_package.rate_reaction_stoichiometry[
+                            r, "Sol", j
+                        ]
+                        for r in solid_phase.reaction_package.rate_reaction_idx
+                    )
+                )
+            else:
+                return b.solid_material_accumulation[t, x, j] == 0
```

Comment:
If there is no `solid_phase.reaction_package`, then the `solid_material_accumulation` is zero. This should be modified to allow modeling cases without reactions such an adsorption with physisorption only. This is the common case for physisorbents like zeolites or traditional MOFs. In those cases the mass balance for solid is represented by a linear or quadratic driving force equation (something like this: dq/dt=k*(q_eq-q))